maven-plugin/build.gradle

```diff
@@ -51,4 +68,64 @@ publishing {
             from components.java
         }
     }
-}
\ No newline at end of file
+}
+
+// Add the mojo descriptor to the test source-set (workaround https://github.com/britter/maven-plugin-development/issues/76)
+sourceSets.test.output.dir file("$buildDir/mavenPlugin/descriptor"), builtBy: generateMavenPluginDescriptor
+
+def genOutputDir = file("$buildDir/generated/test-resources")
+
+task generateTakariTestResources {
+    dependsOn getRootProject().getTasksByName("jar", true)
+    dependsOn getRootProject().getTasksByName("generatePomFileForMavenJavaPublication", true)
+
+    def testProperties = new File(genOutputDir, "test.properties")
+    outputs.file(testProperties)
+
+    def workspaceStateProperties = file("$buildDir/maven-work/workspacestate.properties")
+    outputs.file(workspaceStateProperties)
+
+    doLast {
+        // Generate 'workspacestate.properties' file
+        java.util.Properties workspaceProp = new java.util.Properties();
+        getRootProject().getAllprojects().each { prj -> 
+            def ext = prj.getExtensions().findByType(PublishingExtension.class);
+            if(ext != null) {
+                ext.getPublications().each { pub -> 
+                    addProject(workspaceProp, prj)
+                }
+            }
+        }
+        workspaceProp.store(new java.io.FileOutputStream(workspaceStateProperties), "workspace state file")
+
+        // Generate 'test.properties' file
+        // See https://github.com/takari/takari-plugin-testing-project/blob/master/testproperties.md
+        java.util.Properties testProp = new java.util.Properties();
+
+        testProp.setProperty("project.groupId", getProject().getGroup().toString());
+        testProp.setProperty("project.artifactId", getProject().getConvention().getPlugin(BasePluginConvention.class).getArchivesBaseName());
+        testProp.setProperty("project.version", getProject().getVersion().toString());
+
+        testProp.setProperty("classpath", "" + jar.archiveFile.get().asFile + ":" + configurations.runtimeClasspath.getAsPath());
+
+        testProp.setProperty("localRepository", System.getProperty("user.home") + "/.m2/repository");
+        testProp.setProperty("repository.0", "<id>central</id><url>https://repo.maven.apache.org/maven2</url><releases><enabled>true</enabled></releases><snapshots><enabled>false</enabled></snapshots>");
+        testProp.setProperty("updateSnapshots", "false");
+        // testProp.setProperty("globalSettingsFile", null);
+        File userSettingsFile = new File(System.getProperty("user.home") + "/.m2/settings.xml")
+        if(userSettingsFile.exists()) {
```

Comment:
add a space between `if` and `(`

Reply:
In most of my project I am using a tool like https://github.com/diffplug/spotless, this way the formatting is enforced and all your PR will be properly formatted.

Reply:
```suggestion
        if (userSettingsFile.exists()) {
```